src/widgets/dnshole/DnsHoleControls.tsx

```diff
@@ -124,53 +134,95 @@ function DnsHoleControlsWidgetTile({ widget }: DnsHoleControlsWidgetProps) {
+
+          setTimeout(
+            () => {
+              trpcUtils.dnsHole.summary.invalidate();
+            },
+            (duration + 3) * 1000
+          );
         },
       }
     );
+    setAppId('');
   };
 
   return (
-    <Stack justify="space-between" h={'100%'} spacing="0.25rem">
+    <Stack w={'100%'} spacing="0.25rem">
       {enableControls && widget.properties.showToggleAllButtons && (
-        <SimpleGrid
-          ref={ref}
-          cols={width > 275 ? 2 : 1}
-          verticalSpacing="0.25rem"
-          spacing="0.25rem"
-        >
-          <Button
-            onClick={() => toggleDns('enable', getDnsStatus()?.disabled)}
-            disabled={getDnsStatus()?.disabled.length === 0 || fetchingDnsSummary || changingStatus}
-            leftIcon={<IconPlayerPlay size={20} />}
-            variant="light"
-            color="green"
-            h="2rem"
-          >
-            {t('enableAll')}
-          </Button>
-          <Button
-            onClick={() => toggleDns('disable', getDnsStatus()?.enabled)}
-            disabled={getDnsStatus()?.enabled.length === 0 || fetchingDnsSummary || changingStatus}
-            leftIcon={<IconPlayerStop size={20} />}
-            variant="light"
-            color="red"
-            h="2rem"
-          >
-            {t('disableAll')}
-          </Button>
-        </SimpleGrid>
+        <Flex gap="xs">
+          <Tooltip label={t('enableAll')}>
+            <Button
+              onClick={() => toggleDns('enable', getDnsStatus()?.disabled)}
+              disabled={
+                getDnsStatus()?.disabled.length === 0 || fetchingDnsSummary || changingStatus
+              }
+              variant="light"
+              color="green"
+              fullWidth
+              h="2rem"
+            >
+              <IconPlayerPlay size={20} />
+            </Button>
+          </Tooltip>
+
+          <Tooltip label={t('setTimer')}>
+            <Button
+              onClick={open}
+              disabled={
+                getDnsStatus()?.enabled.length === 0 || fetchingDnsSummary || changingStatus
+              }
+              variant="light"
+              color="yellow"
+              fullWidth
+              h="2rem"
+            >
+              <IconClockPause size={20} />
+            </Button>
+          </Tooltip>
+
+          <Tooltip label={t('disableAll')}>
+            <Button
+              onClick={() => toggleDns('disable', getDnsStatus()?.enabled)}
+              disabled={
+                getDnsStatus()?.enabled.length === 0 || fetchingDnsSummary || changingStatus
+              }
+              variant="light"
+              color="red"
+              fullWidth
+              h="2rem"
+            >
+              <IconPlayerStop size={20} />
+            </Button>
+          </Tooltip>
+        </Flex>
       )}
 
+      <TimerModal
+        toggleDns={toggleDns}
+        getDnsStatus={getDnsStatus}
+        opened={opened}
+        close={close}
```

Comment:
Reset time on closing as well.
You can open the modal, set a time, close it, open again and the set time will still be there.